examples/coin_flipping/coin_flipping.py

```diff
@@ -0,0 +1,172 @@
+import numpy as np
+
+from objects.qubit import Qubit
+from components.host import Host
+from components.network import Network
+from backends.eqsn_backend import EQSNBackend
+# from backends.cqc_backend import CQCBackend
+# from backends.projectq_backend import ProjectQBackend
+
+
+def quantum_coin_flipping(host, m, n, partner_id, rot_angle):
+    """
+    Quantum Coin Flipping Protocol.
+
+    Own random variables are b_j and d_ij.
+    Partners random variables are a_j and c_ij.
+
+    Own shared: f_ij = b_j ^ d_ij
+    Partner shared: e_ij = a_j ^ c_ij
+
+    final own: B = ^b_j for all j
+    final partner: A_tilde = ^a_j for all j
+
+    final random bit: A_tilde ^ B
+    """
+    b = np.random.randint(0, 2, m, dtype=int)
+    d = np.random.randint(0, 2, (n, m), dtype=int)
+
+    a = np.zeros(m, dtype=int)
+
+    # qubits received by partner
+    partner_qubits = np.ndarray(shape=(n, m, 2), dtype=Qubit)
+
+    # qubits which are in the end at host
+    psi_a = np.ndarray(shape=(n, m), dtype=Qubit)
+    psi_b_bar = np.ndarray(shape=(n, m), dtype=Qubit)
+
+    for i in range(n):
+        print(i)
+        for j in range(m):
+            q1 = Qubit(host)
+            q2 = Qubit(host)
+
+            if d[i, j] == 0:
+                q1.rx(rot_angle)
+                q2.rx(-1.0 * rot_angle)
+            else:
+                q1.rx(-1.0 * rot_angle)
+                q2.rx(rot_angle)
+
+            # send and get q1
+            host.send_qubit(partner_id, q1, await_ack=True)
+            partner_q1 = host.get_data_qubit(partner_id)
+
+            # send and get q2
+            host.send_qubit(partner_id, q2, await_ack=True)
+            partner_q2 = host.get_data_qubit(partner_id)
+
+            partner_qubits[i, j, 0] = partner_q1
+            partner_qubits[i, j, 1] = partner_q2
+
+    for i in range(n):
+        print(i)
+        for j in range(m):
+            f_ij = b[j] ^ d[i, j]
```

Comment:
`e_ij, f_ij` is maybe also hard to understand the meaning.

Reply:
I added some comments here